Added info about PlutusV3 to the Read-the-docs page about Plutus language changes (#5844)
IntersectMBO/plutus: doc/read-the-docs-site/reference/cardano/language-changes.rst

Examples
------------

> - `PlutusV2 functionalities <https://github.com/IntersectMBO/cardano-node/blob/master/doc/reference/plutus/babbage-script-example.md>`_

Reviewer: ```suggestion
- `PlutusV2 functionalities <https://github.com/input-output-hk/cardano-node-wiki/blob/main/docs/reference/plutus/babbage-script-example.md>`_
```

All of the built-in types and functions from ``PlutusV1`` were added to ``PlutusV2``.

The following built-in function was added to ``PlutusV2`` only (ie, it is not available in ``PlutusV1``).

- ``serializeData`` (proposed in `CIP-42 <https://cips.cardano.org/cips/cip42/>`_)

PlutusV3
~~~~~~~~~

> Plutus and cryptography teams at IOG, in collaboration with `MLabs <https://mlabs.city/>`_, continue to develop Plutus capabilities. Starting with the release of `Cardano node v.8.8.0-pre <https://github.com/IntersectMBO/cardano-node/releases/tag/8.8.0-pre>`_, ``PlutusV3`` is available on `SanchoNet <https://sancho.network/>`_, introducing the Cardano community to governance features from `CIP-1694 <https://cips.cardano.org/cip/CIP-1694#goal>`_ in a controlled testnet environment. 

Reviewer 2: I suggest removing this paragraph. It's not very useful to talk about pre-releases and sanchonet, especially since `8.8.0-pre` doesn't yet support all the new primitives.

``PlutusV3`` is the new ledger language that enhances Plutus Core's cryptographic capabilities, offering the following benefits for the smart contract developer community: 

- Providing an updated script context that will let users see `CIP-1694 <https://cips.cardano.org/cip/CIP-1694#goal>`_  governance-related entities and voting features

> - Interoperability between blockchains

Reviewer 2: I'd remove these:
```
- Interoperability between blockchains
- Advanced Plutus primitives
- Well-known and optimal cryptographic algorithms
- Support for porting of smart contracts from Ethereum
- Creating sidechain bridges
```

and simply say "new cryptographic and bitwise primitives`. These primitives are already explained below, and it is unclear which specific primitive you are talking about when you say "Creating sidechain bridges" etc.

- Advanced Plutus primitives
- Well-known and optimal cryptographic algorithms
- Support for porting of smart contracts from Ethereum
- Creating sidechain bridges

> - Improving performance by adding a sums of products (SOPs) feature to support the direct encoding of differrent data types.

Reviewer 2: We can just say "sums of products" since it is already explained in the immediate next paragraph.

> The following built-in function was added to ``PlutusV2`` only (i.e., it is not available in ``PlutusV1``).

Reviewer: ```suggestion
The following built-in function was added to ``PlutusV2`` only (ie, it is not available in ``PlutusV1``).
```

- ``serializeData`` (proposed in `CIP-42 <https://cips.cardano.org/cips/cip42/>`_)

PlutusV3
~~~~~~~~~

> Plutus and cryptography teams at IOG, in collaboration with `MLabs <https://mlabs.city/>`_, continue to develop Plutus capabilities. Starting with the release of `Cardano node v8.8.0-pre <https://github.com/IntersectMBO/cardano-node/releases/tag/8.8.0-pre>`_, ``PlutusV3`` is available on `SanchoNet <https://sancho.network/>`_, introducing the Cardano community to governance features from `CIP-1694 <https://cips.cardano.org/cip/CIP-1694#goal>`_ in a controlled testnet environment. 

Reviewer: ```suggestion
Plutus and cryptography teams at IOG, in collaboration with `MLabs <https://mlabs.city/>`_, continue to develop Plutus capabilities. Starting with the release of `Cardano node v.8.8.0-pre <https://github.com/IntersectMBO/cardano-node/releases/tag/8.8.0-pre>`_, ``PlutusV3`` is available on `SanchoNet <https://sancho.network/>`_, introducing the Cardano community to governance features from `CIP-1694 <https://cips.cardano.org/cip/CIP-1694#goal>`_ in a controlled testnet environment. 
```

``PlutusV3`` is the new ledger language that enhances Plutus Core's cryptographic capabilities, offering the following benefits for the smart contract developer community: 

- Providing an updated script context that will let users see `CIP-1694 <https://cips.cardano.org/cip/CIP-1694#goal>`_  governance-related entities and voting features
- Interoperability between blockchains
- Advanced Plutus primitives
- Well-known and optimal cryptographic algorithms
- Support for porting of smart contracts from Ethereum
- Creating sidechain bridges

> - Improving performance by adding a sums of products (SOPs) feature to support the direct encoding of differrent data types

Reviewer: ```suggestion
- Improving performance by adding a sums of products (SOPs) feature to support the direct encoding of differrent data types.
```

Sums of products
~~~~~~~~~~~~~~~~

> ``PlutusV3`` introduces sums of products --- a way of encoding data types that leads to smaller and cheaper scripts compared with `Scott encoding <https://en.wikipedia.org/wiki/Mogensen%E2%80%93Scott_encoding>`_, a common way of encoding data types in Plutus Core. 

Reviewer: ```suggestion
``PlutusV3`` introduces sums of products – a way of encoding data types that leads to smaller and cheaper scripts compared with `Scott encoding <https://en.wikipedia.org/wiki/Mogensen%E2%80%93Scott_encoding>`_, a common way of encoding data types in Plutus Core. 
```

New cryptographic primitives
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

``PlutusV3`` provides new built-in primitives that expand the language's capabilities. 

- **BLS12-381**: A curve pairing that includes 17 primitives that support cryptographic curves. This is a benefit to sidechain specification implementation and `Mithril <https://iohk.io/en/blog/posts/2023/07/20/mithril-nears-mainnet-release/>`_ integration. 
- **Blake2b-224**: A cryptographic hash function for on-chain computation of public-key hashes for the validation of transaction signatures. Supports community projects and contributes to Cardano's versatility. 
- **Keccak-256**: A cryptographic hash function that produces a 256-bit (32-byte) hash value, commonly used for secure data verification. Supports Ethereum signature verification within scripts and cross-chain solutions. 

Bitwise primitives
~~~~~~~~~~~~~~~~~~~

> The introduction of `CIP-58 <https://cips.cardano.org/cip/CIP-0058>`_ bitwise primitives enable the following features: 

Reviewer: ```suggestion
PlutusV3 initially brings several new bitwise primitives (with more to come at later stages). The introduction of `CIP-58 <https://cips.cardano.org/cip/CIP-0058>`_ bitwise primitives will enable the following features: 
```
